src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/SourceControl/VsSourceControlTracker.cs

```diff
@@ -111,20 +110,30 @@ private void StartTracking()
                 return;
             }
 
-            uint cookie;
-            ProjectTracker.AdviseTrackProjectDocumentsEvents(_projectDocumentListener, out cookie);
-            _trackingCookie = cookie;
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                uint cookie;
+                ProjectTracker.AdviseTrackProjectDocumentsEvents(_projectDocumentListener, out cookie);
+                _trackingCookie = cookie;
```

Comment:
>_trackingCookie = cookie [](start = 16, length = 24)

You should check the HRESULT and only regard the out parameters as valid (and therefore assign it to a field) if the HR is a success code.